htdocs/product/class/product.class.php

```diff
@@ -3246,6 +3261,57 @@ public function get_nb_ordersupplier($socid, $mode, $filteronproducttype = -1, $
         return $this->_get_stats($sql, $mode, $year);
     }
 
+    // phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+    /**
+     *  Return nb of units or orders in which product is included
+     *
+     * @param  int    $socid               Limit count on a particular third party id
+     * @param  string $mode                'byunit'=number of unit, 'bynumber'=nb of entities
+     * @param  int    $filteronproducttype 0=To filter on product only, 1=To filter on services only
+     * @param  int    $year                Year (0=last 12 month)
+     * @param  string $morefilter          More sql filters
+     * @return array                            <0 if KO, result[month]=array(valuex,valuey) where month is 0 to 11
+     */
+    public function get_nb_contract($socid, $mode, $filteronproducttype = -1, $year = 0, $morefilter = '')
+    {
+    	// phpcs:enable
+    	global $conf, $user;
+
+    	$sql = "SELECT sum(d.qty), date_format(c.date_contrat, '%Y%m')";
+    	if ($mode == 'bynumber') {
+    		$sql.= ", count(DISTINCT c.rowid)";
+    	}
+    	$sql.= " FROM ".MAIN_DB_PREFIX."contratdet as d, ".MAIN_DB_PREFIX."contrat as c, ".MAIN_DB_PREFIX."societe as s";
+    	if ($filteronproducttype >= 0) {
+    		$sql.=", ".MAIN_DB_PREFIX."product as p";
+    	}
+    	if (!$user->rights->societe->client->voir && !$socid) {
+    		$sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
+    	}
+    	$sql.= " WHERE c.rowid = d.fk_contrat";
+    	if ($this->id > 0) {
```

Comment:
@ptibogxiv 
better performance if you move entity filter in first position

```
$sql.= " WHERE c.entity IN (".getEntity('contract').")";
$sql.= " AND c.rowid = d.fk_contrat";
```

Reply:
@hregis i don't made change on this part (it's actual v11 code) juste copy/paste full code for update old PR

Reply:
@hregis move done ;)